File: apps/api/src/models/attraction.model.ts
/* eslint-disable no-use-before-define */
/* eslint-disable no-shadow */
import { BaseModel } from '@itourism/nestjs';

export class AttractionModel extends BaseModel<AttractionModel> {
  id: string;

  name: string;

  type: string;

  cityId: string;

  latitude: number;

  longitude: number;

  banner: string;

  gallery: string[];

  description: string;

  address: string;

  map: string;

  rate: number;

  createdAt: Date;

  updatedAt: Date;

  static serialize(attraction: AttractionModel): AttractionModel {
    return new AttractionModel({
      id: attraction.id,
      name: attraction.name,
      type: attraction.type,
      cityId: attraction.cityId,
      latitude: attraction.latitude,
      longitude: attraction.longitude,
      banner: attraction.banner,
      gallery: attraction.gallery,
      description: attraction.description,
      address: attraction.address,
      map: attraction.map,
      rate: attraction.rate,
      createdAt: attraction.createdAt,
      updatedAt: attraction.updatedAt,
    });
  }
}

enum attractionType {
  hotel,
  attraction,
  restaurants,
  bars,
}

export type AttractionType = keyof typeof attractionType;

File: apps/mobile/lib/modules/attractions/models/attraction_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'attraction_model.g.dart';

enum AttractionType {
  attraction,
  bars,
  hotel,
  restaurants,
}

@JsonSerializable()
class Attraction {
  String id;

  String name;

  AttractionType type;

  String cityId;

  double latitude;

  double longitude;

  String banner;

  List<String> gallery;

  String description;

  String address;

  String map;

  num rate;

  DateTime createdAt;

  DateTime updatedAt;

  Attraction({
    required this.id,
    required this.name,
    required this.type,
    required this.cityId,
    required this.latitude,
    required this.longitude,
    required this.banner,
    required this.gallery,
    required this.description,
    required this.address,
    required this.map,
    required this.rate,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Attraction.fromJson(Map<String, dynamic> json) =>
      _$AttractionFromJson(json);

  Map<String, dynamic> toJson() => _$AttractionToJson(this);
}
